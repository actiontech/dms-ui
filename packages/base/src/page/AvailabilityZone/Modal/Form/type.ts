import { FormInstance } from 'antd';

export type AvailabilityZoneFormType = {
  name: string;
  address: string;
};

export type AvailabilityZoneFormProps = {
  form: FormInstance<AvailabilityZoneFormType>;
  isUpdate?: boolean;
};
