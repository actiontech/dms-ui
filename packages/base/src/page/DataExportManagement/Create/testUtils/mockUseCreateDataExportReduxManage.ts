import { CreateDataExportPageEnum } from '../../../../store/dataExport';
import { ExportMethodEnum } from '../components/CreateTask/ExportMethodForm/index.enum';
import * as useCreateDataExportReduxManage from '../hooks/index.redux';

export const mockCreateDataExportRedux = {
  updateDataExportInfoOpen: false,
  formValues: {
    baseValues: { workflow_subject: 'test', desc: 'desc' },
    sourceValues: { dbService: '121' },
    methodValues: { sql: 'select 1;', exportMethod: ExportMethodEnum.sql }
  },
  pageState: CreateDataExportPageEnum.CREATE_TASK,
  auditLoading: false,
  submitLoading: false,
  taskIDs: ['11111'],
  workflowID: '123',
  updateFormValues: jest.fn(),
  updatePageState: jest.fn(),
  updateAuditLoading: jest.fn(),
  updateSubmitLoading: jest.fn(),
  initModalStatus: jest.fn(),
  updateModalStatus: jest.fn(),
  updateTaskIDs: jest.fn(),
  updateWorkflowID: jest.fn(),
  clearAllState: jest.fn()
};

export const mockUseCreateDataExportReduxManage = (
  data?: Partial<typeof mockCreateDataExportRedux>
) => {
  const spy = jest.spyOn(useCreateDataExportReduxManage, 'default');
  spy.mockImplementation(() => ({ ...mockCreateDataExportRedux, ...data }));
  return spy;
};
