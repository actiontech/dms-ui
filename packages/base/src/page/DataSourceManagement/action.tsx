import { PlusOutlined } from '@actiontech/icons';
import { ActionButton } from '@actiontech/shared';
import { PermissionControl, PERMISSIONS } from '@actiontech/shared/lib/global';
import { ReactNode } from 'react';
import { DataSourceManagerSegmentedKey } from './index.type';
import { t } from '../../locale';
import { ROUTE_PATHS } from '@actiontech/shared/lib/data/routePaths';

export const DataSourceManagementPageHeaderActions = (
  activeKey: DataSourceManagerSegmentedKey,
  onBatchTestConnection: () => void
): Record<
  | 'add_sync_task'
  | 'batch_import_db_service'
  | 'add_db_service'
  | 'batch_test_data_source_connection',
  ReactNode
> => {
  return {
    add_sync_task: (
      <PermissionControl
        permission={PERMISSIONS.ACTIONS.BASE.SYNC_DATA_SOURCE.ADD}
      >
        <ActionButton
          text={t('dmsSyncDataSource.syncTaskList.addSyncTask')}
          type="primary"
          icon={<PlusOutlined width={10} height={10} color="currentColor" />}
          hidden={activeKey !== DataSourceManagerSegmentedKey.SyncDataSource}
          actionType="navigate-link"
          link={{ to: ROUTE_PATHS.BASE.SYNC_DATA_SOURCE.create }}
        />
      </PermissionControl>
    ),
    batch_import_db_service: (
      <PermissionControl
        permission={PERMISSIONS.ACTIONS.BASE.GLOBAL_DATA_SOURCE.BATCH_IMPORT}
      >
        <ActionButton
          text={t('dmsGlobalDataSource.batchImportDataSource.buttonText')}
          hidden={activeKey !== DataSourceManagerSegmentedKey.GlobalDataSource}
          actionType="navigate-link"
          link={{ to: ROUTE_PATHS.BASE.GLOBAL_DATA_SOURCE.batch_import }}
        />
      </PermissionControl>
    ),
    add_db_service: (
      <PermissionControl
        permission={PERMISSIONS.ACTIONS.BASE.GLOBAL_DATA_SOURCE.ADD}
      >
        <ActionButton
          text={t('dmsGlobalDataSource.addDatabase')}
          type="primary"
          icon={<PlusOutlined width={10} height={10} color="currentColor" />}
          hidden={activeKey !== DataSourceManagerSegmentedKey.GlobalDataSource}
          actionType="navigate-link"
          link={{ to: ROUTE_PATHS.BASE.GLOBAL_DATA_SOURCE.create }}
        />
      </PermissionControl>
    ),
    batch_test_data_source_connection: (
      <PermissionControl
        permission={
          PERMISSIONS.ACTIONS.BASE.GLOBAL_DATA_SOURCE.BATCH_TEST_CONNECT
        }
      >
        <ActionButton
          onClick={onBatchTestConnection}
          hidden={activeKey !== DataSourceManagerSegmentedKey.GlobalDataSource}
          text={t('dmsGlobalDataSource.batchTestDataSourceConnection')}
        />
      </PermissionControl>
    )
  };
};
