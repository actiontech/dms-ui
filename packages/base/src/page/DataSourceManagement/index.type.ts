export enum DataSourceManagerSegmentedKey {
  GlobalDataSource = 'GlobalDataSource',
  SyncDataSource = 'SyncDataSource'
}
