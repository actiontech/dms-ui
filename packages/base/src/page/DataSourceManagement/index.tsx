import { useTranslation } from 'react-i18next';
import {
  PageHeader,
  SegmentedTabs,
  useTypedNavigate,
  useTypedQuery
} from '@actiontech/shared';
import { DataSourceManagerSegmentedKey } from './index.type';
import { useEffect, useMemo, useState } from 'react';
import { Space } from 'antd';
import { TableRefreshButton } from '@actiontech/shared/lib/components/ActiontechTable';
import eventEmitter from '../../utils/EventEmitter';
import EmitterKey from '../../data/EmitterKey';
import GlobalDataSource from '../GlobalDataSource';
import SyncDataSource from '../SyncDataSource';
import { SegmentedTabsProps } from '@actiontech/shared/lib/components/SegmentedTabs/index.type';
import {
  PERMISSIONS,
  PermissionsConstantType
} from '@actiontech/shared/lib/global';
import usePermission from '@actiontech/shared/lib/global/usePermission/usePermission';
import { DataSourceManagementPageHeaderActions } from './action';
import { ROUTE_PATHS } from '@actiontech/shared/lib/data/routePaths';

const DataSourceManagement: React.FC = () => {
  const { t } = useTranslation();
  const [activeKey, setActiveKey] = useState(
    DataSourceManagerSegmentedKey.GlobalDataSource
  );

  const { checkPagePermission } = usePermission();

  const extractQuery = useTypedQuery();
  const navigate = useTypedNavigate();

  const onRefresh = () => {
    if (activeKey === DataSourceManagerSegmentedKey.GlobalDataSource) {
      eventEmitter.emit(EmitterKey.DMS_Refresh_Global_Data_Source);
    } else if (activeKey === DataSourceManagerSegmentedKey.SyncDataSource) {
      eventEmitter.emit(EmitterKey.DMS_Refresh_Sync_Data_Source);
    }
  };

  const tabItems = useMemo<SegmentedTabsProps['items']>(() => {
    const items: Array<
      SegmentedTabsProps['items'][0] & { permission: PermissionsConstantType }
    > = [
      {
        label: t('dmsGlobalDataSource.pageTitle'),
        value: DataSourceManagerSegmentedKey.GlobalDataSource,
        children: <GlobalDataSource />,
        destroyInactivePane: true,
        permission: PERMISSIONS.PAGES.BASE.GLOBAL_DATA_SOURCE
      },
      {
        label: t('dmsSyncDataSource.pageTitle'),
        value: DataSourceManagerSegmentedKey.SyncDataSource,
        children: <SyncDataSource />,
        destroyInactivePane: true,
        permission: PERMISSIONS.PAGES.BASE.SYNC_DATA_SOURCE
      }
    ];

    return items.filter((item) => checkPagePermission(item.permission));
  }, [checkPagePermission, t]);

  // #if [ee]
  const renderExtra = () => {
    const pageHeaderActions = DataSourceManagementPageHeaderActions(activeKey);
    return (
      <>
        {pageHeaderActions['add_sync_task']}
        <Space>
          {pageHeaderActions['batch_import_db_service']}
          {pageHeaderActions['add_db_service']}
        </Space>
      </>
    );
  };
  // #endif

  useEffect(() => {
    const searchParams = extractQuery(
      ROUTE_PATHS.BASE.DATA_SOURCE_MANAGEMENT.index
    );
    if (searchParams && searchParams.active) {
      setActiveKey(searchParams.active as DataSourceManagerSegmentedKey);
    }
  }, [extractQuery]);

  return (
    <article>
      <PageHeader
        title={
          <Space>
            {t('dmsMenu.globalSettings.instanceManager')}
            {/* #if [ee] */}
            <TableRefreshButton refresh={onRefresh} />
            {/* #endif */}
          </Space>
        }
        // #if [ee]
        extra={renderExtra()}
        // #endif
      />

      <SegmentedTabs
        activeKey={activeKey}
        onChange={(key) => {
          setActiveKey(key);
          navigate(ROUTE_PATHS.BASE.DATA_SOURCE_MANAGEMENT.index, {
            queries: { active: key },
            replace: true
          });
        }}
        items={tabItems}
      />
    </article>
  );
};

export default DataSourceManagement;
