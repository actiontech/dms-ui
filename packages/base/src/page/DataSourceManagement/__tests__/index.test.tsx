import { fireEvent } from '@testing-library/dom';
import DataSourceManagement from '..';
import { superRender } from '../../../testUtils/customRender';
import dbServices from '../../../testUtils/mockApi/dbServices';
import syncTaskList from '../../../testUtils/mockApi/syncTaskList';
import { DataSourceManagerSegmentedKey } from '../index.type';
import { useLocation, useNavigate } from 'react-router-dom';
import eventEmitter from '../../../utils/EventEmitter';
import { getBySelector } from '@actiontech/shared/lib/testUtil/customQuery';
import EmitterKey from '../../../data/EmitterKey';
import { mockUsePermission } from '@actiontech/shared/lib/testUtil/mockHook/mockUsePermission';

jest.mock('react-router-dom', () => {
  return {
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
    useLocation: jest.fn()
  };
});

describe('test DataSourceManagement', () => {
  const navigateSpy = jest.fn();

  beforeEach(() => {
    jest.useFakeTimers();
    dbServices.mockAllApi();
    syncTaskList.mockAllApi();
    (useNavigate as jest.Mock).mockImplementation(() => navigateSpy);
    (useLocation as jest.Mock).mockReturnValue({
      pathname: '/data-source-management'
    });
    mockUsePermission({ checkPagePermission: jest.fn().mockReturnValue(true) });
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it('should match snapshot', () => {
    const { container, getByText } = superRender(<DataSourceManagement />);

    expect(container).toMatchSnapshot();
    fireEvent.click(getByText('外部数据源同步'));
    expect(container).toMatchSnapshot();
    expect(navigateSpy).toHaveBeenCalledTimes(1);
    expect(navigateSpy).toHaveBeenCalledWith(
      {
        pathname: '/data-source-management',
        search: `active=${DataSourceManagerSegmentedKey.SyncDataSource}`
      },
      { replace: true }
    );
  });

  it('should send emit event when click refresh icon', () => {
    const emitSpy = jest.spyOn(eventEmitter, 'emit');

    const { getByText } = superRender(<DataSourceManagement />);

    fireEvent.click(getBySelector('.custom-icon-refresh'));

    expect(emitSpy).toHaveBeenCalledTimes(1);
    expect(emitSpy).toHaveBeenCalledWith(
      EmitterKey.DMS_Refresh_Global_Data_Source
    );

    fireEvent.click(getByText('外部数据源同步'));
    fireEvent.click(getBySelector('.custom-icon-refresh'));
    expect(emitSpy).toHaveBeenCalledTimes(2);
    expect(emitSpy).toHaveBeenCalledWith(
      EmitterKey.DMS_Refresh_Sync_Data_Source
    );
  });

  it('should not render tabs item when use role is not admin or not project manager', () => {
    mockUsePermission({
      checkPagePermission: jest.fn().mockReturnValue(false)
    });

    const { queryByText } = superRender(<DataSourceManagement />);

    expect(queryByText('全局数据源')).not.toBeInTheDocument();
    expect(queryByText('外部数据源同步')).not.toBeInTheDocument();
  });
});
