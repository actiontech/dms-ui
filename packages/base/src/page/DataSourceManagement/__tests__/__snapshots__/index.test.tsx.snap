// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DataSourceManagement should match snapshot 1`] = `
<div>
  <article>
    <div
      class="actiontech-page-header-namespace css-1xspr0w"
    >
      <div
        class="title"
      >
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            数据源管理
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-sm ant-btn-icon-only basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  class="custom-icon custom-icon-refresh"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.187 2.586a5.833 5.833 0 0 1 8.59 7.762L9.918 7h1.75a4.667 4.667 0 0 0-7.899-3.367zm7.626 8.828a5.833 5.833 0 0 1-8.59-7.762L4.083 7h-1.75a4.667 4.667 0 0 0 7.899 3.367z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="extra"
      >
        <a
          href="/sync-data-source/create"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
            hidden=""
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <svg
                color="currentColor"
                fill="none"
                height="10"
                viewBox="0 0 10 10"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m5.02.5-.012 9M.5 5h9"
                  stroke="currentColor"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
            <span>
              添加同步任务
            </span>
          </button>
        </a>
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
              type="button"
            >
              <span>
                批量测试数据源连通性
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              href="/global-data-source/batch-import"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                type="button"
              >
                <span>
                  批量导入数据源
                </span>
              </button>
            </a>
          </div>
          <div
            class="ant-space-item"
          >
            <a
              href="/global-data-source/create"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                type="button"
              >
                <span
                  class="ant-btn-icon"
                >
                  <svg
                    color="currentColor"
                    fill="none"
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m5.02.5-.012 9M.5 5h9"
                      stroke="currentColor"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
                <span>
                  添加数据源
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="segmented-tabs-wrapper"
    >
      <div
        class="ant-row segmented-row-wrapper css-156vlto css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-segmented basic-segmented-wrapper css-192b3m1 css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-segmented-group"
          >
            <label
              class="ant-segmented-item ant-segmented-item-selected"
            >
              <input
                checked=""
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
                title="全局数据源"
              >
                全局数据源
              </div>
            </label>
            <label
              class="ant-segmented-item"
            >
              <input
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
                title="外部数据源同步"
              >
                外部数据源同步
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="lazy-load-wrapper segmented-item-content lazy-load-wrapper-show css-952l0m"
      >
        <section>
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-1kh50jg"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-space css-dev-only-do-not-override-1kh50jg ant-space-horizontal ant-space-align-center full-width-element flex-space-between actiontech-table-toolbar-namespace css-c38ntt"
                style="flex-wrap: wrap; margin-bottom: -8px;"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 8px; padding-bottom: 8px;"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-1kh50jg ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 12px;"
                    >
                      <span
                        class="ant-input-affix-wrapper basic-input-wrapper actiontech-table-search-input-namespace css-1f6svkh ant-input-affix-wrapper-sm css-dev-only-do-not-override-mlhibv"
                      >
                        <input
                          class="ant-input ant-input-sm css-dev-only-do-not-override-mlhibv"
                          id="actiontech-table-search-input"
                          placeholder="输入关键字搜索"
                          type="text"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <svg
                            class="custom-icon custom-icon-search"
                            height="14"
                            viewBox="0 0 14 14"
                            width="14"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m10.518 9.693 2.499 2.498-.826.825-2.498-2.498a5.252 5.252 0 0 1-8.527-4.101 5.25 5.25 0 0 1 5.25-5.25 5.252 5.252 0 0 1 4.102 8.526m-1.17-.433a4.082 4.082 0 0 0-2.931-6.927 4.08 4.08 0 0 0-4.084 4.084 4.082 4.082 0 0 0 6.927 2.93z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <button
                        class="ant-btn css-dev-only-do-not-override-czn7jp ant-btn-default ant-btn-sm ant-btn-two-chinese-chars actiontech-filter-button-namespace basic-button-wrapper css-geipcv"
                        style="padding: 0px 6px 0px 10px;"
                        type="button"
                      >
                        <div
                          class="ant-space css-dev-only-do-not-override-czn7jp ant-space-horizontal ant-space-align-center css-1nt49ug"
                        >
                          <div
                            class="ant-space-item"
                            style="margin-right: 2px;"
                          >
                            筛选
                          </div>
                          <div
                            class="ant-space-item"
                          >
                            <svg
                              color="currrentColor"
                              height="14"
                              viewBox="0 0 16 16"
                              width="14"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="m8 8.781 3.3-3.3.943.943L8 10.667 3.757 6.424l.943-.943z"
                                fill="currentColor"
                              />
                            </svg>
                          </div>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                  style="padding-bottom: 8px;"
                >
                  <div
                    class="ant-space css-dev-only-do-not-override-1kh50jg ant-space-horizontal ant-space-align-center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-table-wrapper actiontech-table-namespace css-fxd6ka css-dev-only-do-not-override-bj9uhl"
          >
            <div
              class="ant-spin-nested-loading css-dev-only-do-not-override-bj9uhl"
            >
              <div>
                <div
                  aria-busy="true"
                  aria-live="polite"
                  class="ant-spin ant-spin-spinning css-dev-only-do-not-override-bj9uhl"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
              <div
                class="ant-spin-container ant-spin-blur"
              >
                <div
                  class="ant-table ant-table-empty ant-table-scroll-horizontal"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                      style="overflow-x: auto; overflow-y: hidden;"
                    >
                      <table
                        style="min-width: 100%; table-layout: auto;"
                      >
                        <colgroup />
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              数据源名
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              项目名称
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              环境属性
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              地址
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              上一次连接状态
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              来源
                            </th>
                            <th
                              class="ant-table-cell ellipsis-column-width"
                              scope="col"
                            >
                              描述
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              数据源类型
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              运维时间
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              未完成工单数量
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              工作台查询审核
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            aria-hidden="true"
                            class="ant-table-measure-row"
                            style="height: 0px; font-size: 0px;"
                          >
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                          </tr>
                          <tr
                            class="ant-table-placeholder"
                          >
                            <td
                              class="ant-table-cell"
                              colspan="11"
                            >
                              <div
                                class="ant-table-expanded-row-fixed"
                                style="width: 0px; position: sticky; left: 0px; overflow: hidden;"
                              />
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </article>
</div>
`;

exports[`test DataSourceManagement should match snapshot 2`] = `
<div>
  <article>
    <div
      class="actiontech-page-header-namespace css-1xspr0w"
    >
      <div
        class="title"
      >
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            数据源管理
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-sm ant-btn-icon-only basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  class="custom-icon custom-icon-refresh"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.187 2.586a5.833 5.833 0 0 1 8.59 7.762L9.918 7h1.75a4.667 4.667 0 0 0-7.899-3.367zm7.626 8.828a5.833 5.833 0 0 1-8.59-7.762L4.083 7h-1.75a4.667 4.667 0 0 0 7.899 3.367z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="extra"
      >
        <a
          href="/sync-data-source/create"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <svg
                color="currentColor"
                fill="none"
                height="10"
                viewBox="0 0 10 10"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m5.02.5-.012 9M.5 5h9"
                  stroke="currentColor"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
            <span>
              添加同步任务
            </span>
          </button>
        </a>
        <div
          class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
              hidden=""
              type="button"
            >
              <span>
                批量测试数据源连通性
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            <a
              href="/global-data-source/batch-import"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
                hidden=""
                type="button"
              >
                <span>
                  批量导入数据源
                </span>
              </button>
            </a>
          </div>
          <div
            class="ant-space-item"
          >
            <a
              href="/global-data-source/create"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
                hidden=""
                type="button"
              >
                <span
                  class="ant-btn-icon"
                >
                  <svg
                    color="currentColor"
                    fill="none"
                    height="10"
                    viewBox="0 0 10 10"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m5.02.5-.012 9M.5 5h9"
                      stroke="currentColor"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
                <span>
                  添加数据源
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="segmented-tabs-wrapper"
    >
      <div
        class="ant-row segmented-row-wrapper css-156vlto css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-segmented basic-segmented-wrapper css-192b3m1 css-dev-only-do-not-override-txh9fw"
        >
          <div
            class="ant-segmented-group"
          >
            <label
              class="ant-segmented-item"
            >
              <input
                checked=""
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
                title="全局数据源"
              >
                全局数据源
              </div>
            </label>
            <label
              class="ant-segmented-item ant-segmented-item-selected"
            >
              <input
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
                title="外部数据源同步"
              >
                外部数据源同步
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="lazy-load-wrapper segmented-item-content lazy-load-wrapper-show css-952l0m"
      >
        <article>
          <div
            class="ant-table-wrapper actiontech-table-namespace css-fxd6ka css-dev-only-do-not-override-bj9uhl"
          >
            <div
              class="ant-spin-nested-loading css-dev-only-do-not-override-bj9uhl"
            >
              <div>
                <div
                  aria-busy="true"
                  aria-live="polite"
                  class="ant-spin ant-spin-spinning css-dev-only-do-not-override-bj9uhl"
                >
                  <span
                    class="ant-spin-dot ant-spin-dot-spin"
                  >
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                    <i
                      class="ant-spin-dot-item"
                    />
                  </span>
                </div>
              </div>
              <div
                class="ant-spin-container ant-spin-blur"
              >
                <div
                  class="ant-table ant-table-empty ant-table-scroll-horizontal"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                      style="overflow-x: auto; overflow-y: hidden;"
                    >
                      <table
                        style="min-width: 100%; table-layout: auto;"
                      >
                        <colgroup />
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              同步任务名
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              来源
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              地址
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              数据源类型
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              最后一次同步结果
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              最近一次同步成功时间
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            aria-hidden="true"
                            class="ant-table-measure-row"
                            style="height: 0px; font-size: 0px;"
                          >
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                          </tr>
                          <tr
                            class="ant-table-placeholder"
                          >
                            <td
                              class="ant-table-cell"
                              colspan="6"
                            >
                              <div
                                class="ant-table-expanded-row-fixed"
                                style="width: 0px; position: sticky; left: 0px; overflow: hidden;"
                              />
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </div>
    </div>
  </article>
</div>
`;
